src/argilla/training/peft.py

```diff
@@ -0,0 +1,256 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import List, Union
+
+import numpy as np
+
+import argilla as rg
+from argilla.training.transformers import ArgillaTransformersTrainer
+from argilla.training.utils import (
+    filter_allowed_args,
+)
+from argilla.utils.dependency import require_version
+
+
+class ArgillaTransformersPEFTTrainer(ArgillaTransformersTrainer):
```

Comment:
Even though `peft` works on top of `transformers`, shouldn't it be better to name this `ArgillaPEFTTrainer` instead, or are we keeping the `Transformers` due to the class-inheritance? IMO we can be good with just `ArgillaPEFTTrainer`, but up to you 😄

Reply:
I agree